modis_tools/granule_handler.py

```diff
@@ -168,11 +173,20 @@ def _get(
         return req
 
     @staticmethod
-    def _get_location(url: HttpUrl, session: Session) -> str:
+    def _get_location(
+        url: HttpUrl, session: Session, modis_session: ModisSession
+    ) -> str:
         """Make initial request to fetch file location from header."""
         split_result = urlsplit(url)
         https_url = split_result._replace(scheme="https").geturl()
         location_resp = session.get(https_url, allow_redirects=False)
+        if location_resp.status_code == 401:
+            # try using ProxyAuth if BasicAuth returns 401 (unauthorized)
+            location_resp = session.get(
+                https_url,
+                allow_redirects=False,
+                auth=HTTPProxyAuth(modis_session.username, modis_session.password),
+            )
```

Comment:
Can confirm that without this snippet the code fails with the error you uncovered before 
(No file location found). Well done